Compiler/script/cc_macrotable.h

```diff
@@ -9,9 +9,12 @@ struct MacroTable {
     char*macro[MAXDEFINES];
     void init();
     void shutdown();
-    int  find_name(char*);
-    void add(char*,char*);
+    int  find_name(const char*);
+    bool contains(const char *);
+    char* get_macro(const char *);
```

Comment:
I think "get_macro" should also return `const char*` if it's for reading. For editing, I would not recommend such function, because it returns a pointer to *some* buffer, but without telling its size nor letting to increase it.

Reply:
Also, tbh, the whole MacroTable struct could be replaced with a std::map<string, string> probably.

I may be mistaken here, but it looks like it's not used anywhere but the preprocessor, which you are rewriting from scratch anyway; if that's correct, then replacing with std::map won't be difficult to do.

Reply:
so, unordered_map gives me all sort of weird errors with the hashing function and ags String type - works fine with std::strings. Using map it works fine with AGS Strings, is map fine, or it HAS to be unordered map?

Reply:
I forgot to mention, we have to provide a hashing function for a String class. It's in utils/string_types.h, simply including it should work.

As for selection between ordered and unordered, iirc ordered may be better when you constantly adding and removing items, and unordered is better when you add them once and then mostly keep looking up.